Lib/test/test_random.py

```diff
@@ -88,6 +89,25 @@ def test_shuffle(self):
         self.assertTrue(lst != shuffled_lst)
         shuffle(lst)
         self.assertTrue(lst != shuffled_lst)
+        self.assertRaises(TypeError, shuffle, (1, 2, 3))
+        self.assertRaises(TypeError, shuffle, {1, 2, 3})
+        self.assertRaises(TypeError, shuffle, 'shuffle')
+        self.assertRaises(KeyError, shuffle, dict(a=1, b=2))
+
+    @unittest.expectedFailure
+    def test_shuffle_dict_with_numeric_keys(self):
+        shuffle = self.gen.shuffle
+        seq = dict(zip(range(9), 'abcdefghi'))
+        self.assertRaises(KeyError, shuffle, seq)
+
+    def test_shuffle_random_argument(self):
+        # Test random argument to shuffle.
+        shuffle = self.gen.shuffle
+        mock_random = unittest.mock.Mock(return_value=0.5)
+        seq = bytearray(b'abcdefghijk')
+        shuffle(seq, mock_random)
+        self.assertEqual(seq, b'ajbhcgdiekf')
```

Comment:
Let's drop line 169.  We're not guaranteeing a particular shuffle order.  Instead, we're just checking the shuffle worked at all when given a shuffle argument.